go/ct/rlz/domains_test.go

```diff
@@ -79,3 +85,296 @@ func TestRemoveDuplicatesGeneric(t *testing.T) {
+			if tc.got != tc.want {
+				t.Errorf("want %v, but got %v", tc.want, tc.got)
+			}
+		})
+	}
+}
+
+func TestDomain_PredecessorPanic(t *testing.T) {
+
+	tests := map[string]struct {
+		value        interface{}
+		generic_type reflect.Type
+	}{
+		"bool":       {value: true, generic_type: reflect.TypeOf(true)},
+		"statusCode": {value: st.Running, generic_type: reflect.TypeOf(st.Running)},
+		"opCode":     {value: vm.ADD, generic_type: reflect.TypeOf(vm.ADD)},
+	}
+
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			defer func() {
+				if r := recover(); r == nil {
+					t.Errorf("The code did not panic")
+				}
+			}()
+			switch tc.generic_type {
+			case reflect.TypeOf(true):
+				boolDomain{}.Predecessor(tc.value.(bool))
+			case reflect.TypeOf(st.Running):
+				statusCodeDomain{}.Predecessor(tc.value.(st.StatusCode))
+			case reflect.TypeOf(vm.ADD):
+				opCodeDomain{}.Predecessor(tc.value.(vm.OpCode))
+			default:
+				t.Errorf("Add type to test cases: %v", tc.generic_type)
+			}
+		})
+	}
+
+}
+
+func TestDomain_Predecessor(t *testing.T) {
+
+	tests := map[string]struct {
+		got  interface{}
+		want interface{}
+	}{
+		"uint16": {got: uint16Domain{}.Predecessor(uint16(1)), want: uint16(0)},
+		"revision": {got: revisionDomain{}.Predecessor(tosca.R09_Berlin),
+			want: tosca.R07_Istanbul},
+		"revision-istanbul": {got: revisionDomain{}.Predecessor(tosca.R07_Istanbul),
+			want: common.R99_UnknownNextRevision},
+		"revision-unknown": {got: revisionDomain{}.Predecessor(common.R99_UnknownNextRevision),
+			want: common.NewestSupportedRevision},
+		"pc": {got: pcDomain{}.Predecessor(common.NewU256(1, 2, 3, 4)),
+			want: common.NewU256(1, 2, 3, 3)},
+		"pc-0": {got: pcDomain{}.Predecessor(common.NewU256(1, 2, 3, 0)),
+			want: common.NewU256(1, 2, 2, 0xffffffffffffffff)},
+		"stackSize":         {got: stackSizeDomain{}.Predecessor(1), want: 0},
+		"blocknumberoffset": {got: BlockNumberOffsetDomain{}.Predecessor(int64(1)), want: int64(0)},
+		"u256": {got: u256Domain{}.Predecessor(common.NewU256(1, 2, 3, 4)),
+			want: common.NewU256(1, 2, 3, 3)},
+		"u256-0": {got: u256Domain{}.Predecessor(common.NewU256(1, 2, 0, 4)),
+			want: common.NewU256(1, 2, 0, 3)},
+	}
+
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			if tc.got != tc.want {
+				t.Errorf("want %v, but got %v", tc.want, tc.got)
+			}
+		})
+	}
+}
+
+func TestDomain_SuccessorPanic(t *testing.T) {
+
+	tests := map[string]struct {
+		value        interface{}
+		generic_type reflect.Type
+	}{
+		"bool":       {value: true, generic_type: reflect.TypeOf(true)},
+		"statusCode": {value: st.Running, generic_type: reflect.TypeOf(st.Running)},
+		"opCode":     {value: vm.ADD, generic_type: reflect.TypeOf(vm.ADD)},
+	}
+
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			defer func() {
+				if r := recover(); r == nil {
+					t.Errorf("The code did not panic")
+				}
+			}()
+			switch tc.generic_type {
```

Comment:
For situations like this, Go has a [type-switch](https://go.dev/tour/methods/16) statement. With this, this test case could be shortened to:
```

func TestDomain_PredecessorPanic(t *testing.T) {
	tests := map[string]any{
		"bool":       true,
		"statusCode": st.Running,
		"opCode":     vm.ADD,
	}

	for name, value := range tests {
		t.Run(name, func(t *testing.T) {
			defer func() {
				if r := recover(); r == nil {
					t.Errorf("The code did not panic")
				}
			}()
			switch value := value.(type) {
			case bool:
				boolDomain{}.Predecessor(value)
			case st.StatusCode:
				statusCodeDomain{}.Predecessor(value)
			case vm.OpCode:
				opCodeDomain{}.Predecessor(value)
			default:
				t.Errorf("Add type to test cases: %v", reflect.TypeOf(value))
			}
		})
	}
}
```